codeserver-python/rockcraft.yaml

```
@@ -0,0 +1,246 @@
+name: codeserver-python
+summary: An image for using Codeserver & Jupyter on CPUs
+description: |
+  This image is used as part of the Charmed Kubeflow product. It is deployed
+  in a user namespace, and enables users to perform CPU-accelerated tasks
+  using PyTorch from within a Jupyter Notebook.
+
+  Both PyTorch and Jupyter are installed in Conda environment, which is
+  automatically activated.
+version: "1.10.0-rc.1"
+license: Apache-2.0
+base: ubuntu@22.04
+run-user: _daemon_
+
+services:
+  codeserver-jupyter:
+    override: replace
+    summary: "codeserver-jupyter service"
+    startup: enabled
+    command: bash -c '/usr/bin/code-server --bind-addr 0.0.0.0:8888 --disable-telemetry --disable-update-check --disable-workspace-trust --disable-getting-started-override --auth none --trusted-origins "*" ${HOME}'
+    working-dir: /home/jovyan/
+    environment:
+       NB_USER: jovyan
+       NB_UID: 1000
+       NB_GID: 0
+       NB_PREFIX: /
+       USERS_GID: 100
+       HOME: /home/jovyan
+       HOME_TMP: /tmp_home/jovyan
+       SHELL: /bin/bash
+       LANG: en_US.UTF-8
+       LANGUAGE: en_US.UTF-8
+       LC_ALL: en_US.UTF-8
+       PATH: /opt/conda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+       PYTHONPATH: /opt/conda/lib/python3.11/site-packages/
+       XDG_RUNTIME_DIR: /tmp/runtime-${NB_USER}
+platforms:
+  amd64:
+
+parts:
+  security-team-requirement:
+    plugin: nil
+    override-build: |
+      mkdir -p ${CRAFT_PART_INSTALL}/usr/share/rocks
+      (echo "# os-release" && cat /etc/os-release && echo "# dpkg-query" && \
+      dpkg-query -f '${db:Status-Abbrev},${binary:Package},${Version},${source:Package},${Source:Version}\n' -W) \
+      > ${CRAFT_PART_INSTALL}/usr/share/rocks/dpkg.query
+
+  overlay-pkgs:
+    plugin: nil
+    overlay-packages:
+      - apt-transport-https
+      - bash
+      - bzip2
+      - ca-certificates
+      - curl
+      - git
+      - gnupg
+      - gnupg2
+      - locales
+      - lsb-release
+      - nano
+      - software-properties-common
+      - tzdata
+      - unzip
+      - vim
+      - wget
+      - xz-utils
+      - zip
+
+  kubectl:
+    plugin: nil
+    stage-snaps:
+      - kubectl/1.31/stable
+    organize:
+      kubectl: bin/kubectl
+    stage:
+      - bin/kubectl
+
+  nodejs:
+    plugin: nil
+    stage-packages: [nodejs]
+
+  generate-locale:
+    plugin: nil
+    build-packages:
+      - locales
+    override-build: |
+      echo "en_US.UTF-8 UTF-8" > /etc/locale.gen
+      locale-gen
+      mkdir -p $CRAFT_PART_INSTALL/usr/lib/locale $CRAFT_PART_INSTALL/usr/share/i18n/locales
+      cp /usr/lib/locale/locale-archive $CRAFT_PART_INSTALL/usr/lib/locale/locale-archive
+      cp -r /usr/share/i18n/locales $CRAFT_PART_INSTALL/usr/share/i18n/locales
+
+  conda-codeserver:
+    plugin: nil
+    source: https://github.com/kubeflow/kubeflow
+    source-type: git
+    source-tag: v1.10.0-rc.1
+    build-packages:
+      - lsb-release
+      - wget
```

We should add the `libxml2` package, otherwise I get the following error:
```
:: Error while loading conda entry point: conda-libmamba-solver (libxml2.so.2: cannot open shared object file: No such file or directory)
```
```suggestion
      - lsb-release
      - wget
```

---

This will not solve the issue. It works when `--solver=classic` is provided for `conda install ipykernel`.